### DISCUSSION TITLE
How to extract data from image using semantic kernal with AzureChatCompletion

### DISCUSSION THREAD

**Discussion by Aswin4070** [microsoft/semantic-kernel#6692] [Q&A] [answered]
Hi everyone,

i am trying a project to extract data from image with the help of gpt4o.
I am able to do this with Azure Openai, but not able to do with semantic kernal.

Need help to develop this. When i tried to look at the documentation, i could see there is a library called ImageContent for dotnet:
https://github.com/microsoft/semantic-kernel/blob/main/dotnet/src/SemanticKernel.Abstractions/Contents/ImageContent.cs

but i cant see anything for python. So is there any other alternative available for python for data extraction from image?

**Comment by jamesmkfoo23** [comment]
I'm trying to do the same. How do i call SK plugins and pass the image (base64 or url) parameter?

**Reply by jamesmkfoo23** [reply]
https://youtu.be/ZaCjAKLMhYg?t=457
This seems to provide a way to add ImageContent to ChatMessageContentItemCollection in .NET. How about in Python?

**Comment by moonbox3** [comment, marked as answer]
Hi @Aswin4070, @jamesmkfoo23, we released the ability to do this in Python today (6/13/24) in v1.1.0. Our docs aren't updated to reflect this yet; however, you can find [a concept sample](https://github.com/microsoft/semantic-kernel/blob/main/python/samples/concepts/chat_completion/azure_chat_image_input.py) showing how to work with this type of content.

**Reply by jamesmkfoo23** [reply]
Thanks @moonbox3 ! I got it working. Is there a away to specify the quality similar to how the REST API allows to?
<img width="391" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/45367108/5bd18334-f7fb-4a2f-9338-b9037d64818a">

**Reply by moonbox3** [reply]
Hi @jamesmkfoo23 apologies -- this property was missed and we will get it added early next week.

**Reply by jamesmkfoo23** [reply]
No apologies needed @moonbox3 ! Appreciate your speedy response and I'm truly impressed!
Thank you.
